src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -308,7 +308,19 @@ func (r *Cluster) KafkaTLSListener() *KafkaAPIListener {
 // AdminAPIInternal returns internal admin listener
 func (r *Cluster) AdminAPIInternal() *AdminAPI {
 	for _, el := range r.Spec.Configuration.AdminAPI {
-		return &el
+		if !el.External.Enabled {
+			return &el
+		}
+	}
+	return nil
+}
+
+// AdminAPIExternal returns internal admin listener
+func (r *Cluster) AdminAPIExternal() *AdminAPI {
+	for _, el := range r.Spec.Configuration.AdminAPI {
+		if !el.External.Enabled {
```

Comment:
Should this be a positive condition - external enabled?

func comment: internal => external

Reply:
thanks, fixed